### PR TITLE
[SYCL][XPTI] Add USM calls parameter verification layer to sycl-trace

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**
1) Extended USM calls verified in sycl-sanitize
2) Add this analysis to sycl-trace (new mode, paired with pi mode only) tool with no termination of process
3) Fixed code location handling in sycl-sanitize/sycl-trace for USM verification
Example of print is:
$ sycl-sanitize ./a.out 
[USM] Function uses nullptr as kernel parameter with index = 0.
      | kernel location:  function runKernel at ./main.cpp:24
cmd line: sycl-trace --verify --sycl ./a.out

### CODE REVIEW

**Review by tovinkere — CHANGES_REQUESTED** [review]
Need clarification on trace types chosen in emitFunctionWithArgsBeginTrace()

**Comment by KseniyaTikhomirova** [comment]
> sycl/source/detail/pi.cpp

@tovinkere could you please advise how to improve that? I need to add event for every call to be able to pass code_location (it could be properly passed only by event) and I took graph event. Would it be ok if I create new event type like "trace_diagnostic_event" as a general purpose event?

File: sycl/source/detail/pi.cpp
Comment (KseniyaTikhomirova) [inline sycl/source/detail/pi.cpp:163]: to check if we need to update xpti documentation

Comment (KseniyaTikhomirova) [reply sycl/source/detail/pi.cpp:163]: https://github.com/intel/llvm/pull/9840/commits/dbee52f5456e4aed56d60986abd2b49bb1f1ad9f

**Review by tovinkere — APPROVED** [review]
LGTM

**Comment by KseniyaTikhomirova** [comment]
@tovinkere I added small fixes in https://github.com/intel/llvm/pull/9840/commits/5d58f894b81a4f817bb31d2d7296a5f8a4ebf74e. Could you please review it? (small change)

**Comment by KseniyaTikhomirova** [comment]
Failed Tests (1):
SYCL :: ESIMD/accessor_local.cpp - Reported in https://github.com/intel/llvm/issues/10138.

**Review by tovinkere — COMMENTED** [review]
LGTM

**Comment by KseniyaTikhomirova** [comment]
Unexpectedly Passed Tests (1):
SYCL :: DeviceArchitecture/device_architecture_on_host.cpp - Seems that our drivers on pre-commit machines are ready for this one to be enabled. Tag @dm-vodopyanov for awareness.

**Comment by KseniyaTikhomirova** [comment]
@intel/llvm-gatekeepers hi, this PR is ready for merge

**Comment by dm-vodopyanov** [comment]
> Unexpectedly Passed Tests (1): SYCL :: DeviceArchitecture/device_architecture_on_host.cpp - Seems that our drivers on pre-commit machines are ready for this one to be enabled. Tag @dm-vodopyanov for awareness.

https://github.com/intel/llvm/commit/a54d2a2c3039541351e9535303ca3d2a1f7f6f19

**Comment by dm-vodopyanov** [comment]
> @intel/llvm-gatekeepers hi, this PR is ready for merge

Code owner review required
Waiting on code owner review from intel/llvm-reviewers-runtime.

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by steffenlarsen** [comment]
@KseniyaTikhomirova - This introduced some post-commit failures. Could you please address these ASAP?

**Comment by KseniyaTikhomirova** [comment]
https://github.com/intel/llvm/pull/10206 should fix post commit

**Pull request merged**
